backend/server/src/db/model/drv.rs

```diff
@@ -66,6 +68,19 @@ impl DrvId {
             None => path,
         }
     }
+
+    /// Take a drv store path, and make it into a DrvId
+    pub fn new(drv: String) -> Self {
+        DrvId(drv.strip_prefix("/nix/store/").unwrap_or(&drv).to_string())
+    }
+
+    pub fn drv_id(&self) -> &str {
```

Comment:
You should be able to just call any `str` methods on a `DrvId` and if coercion does not work you can call `.deref` to get the `&str` (it comes from the `Deref` impl above)